### PR TITLE
[MXNET-16] move nnvm & dmlc-core to 3rdparty

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by eric-haibin-lin** [comment]
@rahul003 could you help review changes in launch.py?
@mbaijal could you help review changes related to license?
@szha could you help review changes in setup.py?
@cjolivier01 @yajiedesign could you help review make/cmake?
@sergeykolychev could you help review changes for perl?
@marcoabreu could you help review changes for Jenkins?

File: tools/launch.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 import logging
 
 curr_path = os.path.abspath(os.path.dirname(__file__))
-sys.path.append(os.path.join(curr_path, "../dmlc-core/tracker"))
+sys.path.append(os.path.join(curr_path, "..3rdparty/dmlc-core/tracker"))
```
Comment (rahul003) [inline tools/launch.py]: I think there should be a slash before 3rdparty?

File: amalgamation/amalgamation.py [hunk 2]
```diff
@@ -149,8 +154,8 @@ def expand(x, pending, stage):
 expand.fileCount = 0
 
 # Expand the stages
-expand(sys.argv[2], [], "dmlc")
-expand(sys.argv[3], [], "nnvm")
+expand(sys.argv[2], [], "3rdparty/dmlc")
```
Comment (szha) [inline amalgamation/amalgamation.py]: what's a stage in expand()? this doesn't match the existing path. is this intended?

**Comment by sergeykolychev** [comment]
@eric-haibin-lin perl change looks good to me.

File: .gitmodules [hunk 3]
```diff
@@ -2,13 +2,13 @@
 	path = 3rdparty/mshadow
 	url = https://github.com/dmlc/mshadow.git
 [submodule "dmlc-core"]
```
Comment (rahul003) [inline .gitmodules]: why are some submodules like "3rdparty/openmp" and others are "ps-lite"? Is this just an identifier? Even so, should we standardize this?

Comment (eric-haibin-lin) [reply .gitmodules]: Good question. The name `ps-lite` remains unchanged when I use the `git mv` command to move the submodule to another folder. I agree that it makes sense to standardize. Let me try to update the name locally first

File: amalgamation/amalgamation.py (hunk 2 above)
Comment (eric-haibin-lin) [reply amalgamation/amalgamation.py]: The `stage` is used to match the prefix of a file. Updating it to `dmlc-core` won't affect the result. Let me document the usage.

**Comment by eric-haibin-lin** [comment]
@sergeykolychev `test_dynamic ` test for perl failed in the most recent commit. Do you think that's caused by this PR?

**Comment by eric-haibin-lin** [comment]
The CI is not running. I'm going to close it and create a new one.

**Comment by TaoLv** [comment]
@eric-haibin-lin How can I merge these two folders? I am trying to merge master branch into my local dev branch.
```
# Unmerged paths:
#   (use "git add <file>..." to mark resolution)
#
#       added by us:        3rdparty/dmlc-core
#       added by us:        3rdparty/nnvm
```

**Pull request closed without merging**